tests/templates/y_cable_simulator_client.j2

```diff
@@ -16,6 +21,129 @@ helper_logger = logger.Logger(SYSLOG_IDENTIFIER)
 UPPER_TOR = "upper_tor"
 LOWER_TOR = "lower_tor"
 
+PLATFORM_JSON = 'platform.json'
+PORT_CONFIG_INI = 'port_config.ini'
+
+# A dict for mapping physical port to host interface
+g_physical_to_host_port_mapping = {}
+g_physical_to_logical_port_mapping = {}
+
+def _physical_port_to_host_port(physical_port):
+    """
+    Convert physical port to host interface index.
+    The physical port index got from spf is different with host interface index when interface is splited.
+    However, the mux bridges on vm_host are named according to host interface index (vm_topology.py).
+    So a convert is needed.
+
+    @arg physical_port: The physical port index got from sfputil
+    @returns: The host interface index (0-based)
+    """
+    global g_physical_to_host_port_mapping
+    global g_physical_to_logical_port_mapping
+
+    if not g_physical_to_host_port_mapping or not g_physical_to_logical_port_mapping:
+        _load_port_info()
+    # Return physical_port - 1 in case loading port config file failure
+    return g_physical_to_host_port_mapping.get(physical_port, physical_port - 1)
+
+def _load_port_info():
+    """
+    Parse platform.json or port_config.ini to get the mapping between physical_port and host interface index
+    and physical_port and logical port name. The mapping is saved in two global variables.
+    """
+    porttabfile = device_info.get_path_to_port_config_file()
+    parse_fmt_platform_json = (os.path.basename(porttabfile) == PLATFORM_JSON)
+
+    if parse_fmt_platform_json:
+        _load_port_config_json()
+    else:
+        _load_port_config_ini(porttabfile)
+
+def _load_port_config_json():
+    """
+    A helper function for loading port config from 'platform.json'
+    """
+    global g_physical_to_host_port_mapping
+    global g_physical_to_logical_port_mapping
+
+    (platform, hwsku) = device_info.get_platform_and_hwsku()
+    ports, _, _ = get_port_config(hwsku, platform)
+    if not ports:
+        helper_logger.log_warning('Failed to get port config')
+        return
+    else:
+        logical_list = []
+        for intf in ports.keys():
+            logical_list.append(intf)
+
+        logical = natsorted(logical_list, key=lambda y: y.lower())
```

Comment:
How about using a single line of code for the purpose of line 75-79:
```
logical = natsorted([intf.lower() for intf in ports.keys()])
```

Reply:
Using a single line code is a good suggestion. But I'd like to keep current code here because converting all interface name into lower case can cause some issue in following code.